File: src/main/java/com/pretty/platform/catalog/domain/model/valueobjects/Category.java
package com.pretty.platform.catalog.domain.model.valueobjects;

import jakarta.persistence.Embeddable;

/**
 * Value object for product category
 */
@Embeddable
public record Category(String name) {
    public Category {
        if (name == null || name.trim().isEmpty()) {
            throw new IllegalArgumentException("Category name cannot be null or empty");
        }
    }
}

File: internal/products/domain/model/valueobjects/category.go
package valueobjects

import "errors"

type Category struct {
	value string `gorm:"column:category"`
}

func NewCategory(value string) (Category, error) {
	if value == "" {
		return Category{}, errors.New("category cannot be empty")
	}
	return Category{value: value}, nil
}

func (c Category) Value() string {
	return c.value
}

func (c Category) String() string {
	return c.value
}
